hive-bigquery-connector-common/src/main/java/com/google/cloud/hive/bigquery/connector/utils/hive/HiveUtils.java

```diff
@@ -73,9 +85,39 @@ private static JobID getJobIDWithVertexAppended(JobID jobID, String vertexId) {
     }
   }
 
-  private static class TaskAttemptIDWrapper extends TaskAttemptID {
+  public static String getTaskID(Configuration conf) {
+    if (isSparkJob(conf)) {
+      return getSparkTaskID();
+    }
+    return getHiveTaskAttemptIDWrapper(conf).getTaskID().toString();
+  }
+
+  /**
+   * Retrieves the Spark task ID from the Spark context. Uses reflection to avoid this library
+   * requiring dependencies on Spark packages.
+   */
+  public static String getSparkTaskID() {
+    try {
+      Class<?> taskContextClass = Class.forName("org.apache.spark.TaskContext");
+      Method getMethod = taskContextClass.getMethod("get");
+      Object taskContext = getMethod.invoke(null);
+      Method taskAttemptIdMethod = taskContextClass.getMethod("taskAttemptId");
```

Comment:
we want task id not task attempt id here?

Reply:
we are only getting task id for hive. this remind that we need to set readme that spark/hive speculative is not supported.

Reply:
@yigress Are you sure? It looks like we're using the attempt id, not just the task id: https://github.com/GoogleCloudDataproc/hive-bigquery-connector/blob/main/hive-bigquery-connector-common/src/main/java/com/google/cloud/hive/bigquery/connector/utils/hive/HiveUtils.java#L63-L105

Reply:
we are writing output file to be just task id. JobUtils#getTaskWriterOutputFile. here we can have the task attempt id just need to make sure the taskid can be fetched from it.

Reply:
this works if the spark application just fails when the insert query fails. however if user choose not to fail the application (for example try catch the query failure) or using spark-shell, it will be the same spark application, if user do another query either insert into same table or insert into another table, what would happen if the spark job info file already exists?